### PR TITLE
OGM-534 Add support for count in JP-QL

### PR DESCRIPTION
**Pull request by Krerg**
https://hibernate.atlassian.net/browse/OGM-534

### CODE REVIEW

**Comment by Krerg** [comment]
@gsmet, Done!

File: core/src/test/java/org/hibernate/ogm/backendtck/queries/AggregateOperationQueryTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * @author Aleksandr Mylnikov
+ */
+public class AggregateOperationQueryTest extends OgmJpaTestCase {
+
+	private EntityManager em;
+
+	@Test
+	@SkipByGridDialect(
+			value = { HASHMAP, INFINISPAN, INFINISPAN_REMOTE, NEO4J_EMBEDDED, NEO4J_REMOTE },
```
Comment (fax4ever) [inline core/src/test/java/org/hibernate/ogm/backendtck/queries/AggregateOperationQueryTest.java]: I think we shall move the `@SkipByGridDialect` annotation on class.

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java [hunk 2]
```diff
@@ -245,6 +251,7 @@ public MongoDBQueryDescriptor build() throws NativeQueryParseException {
 						null,
 						null,
 						null
+
```
Comment (fax4ever) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java]: Could you remove the white space?

**Comment by Krerg** [comment]
@fax4ever Done!

**Comment by Krerg** [comment]
@DavideD , could you please review my changes? I fixed all remarks

**Comment by DavideD** [comment]
Thanks, I will start the review after lunch

**Comment by Krerg** [comment]
@DavideD , Is your lunch over? :blush:

**Comment by Sanne** [comment]
ah! sorry @Krerg , Davide is away travelling.

@fax4ever could you take over the review? Thanks!

**Comment by DavideD** [comment]
> @DavideD , Is your lunch over? blush

It was an Italian lunch, I'm waiting for the dessert :D

Joking aside, sorry about it. I got confused by the label that still says: `changes required`

File: mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java [hunk 3]
```diff
@@ -157,6 +158,11 @@ public boolean setOrderBy(String orderBy) {
 		return true;
 	}
 
+	public boolean setCount(boolean count) {
+		this.count = count;
+		return true;
+	}
+
```
Comment (DavideD) [inline mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java]: @Krerg One quick question: Is this field still necessary?
(Non need to update the PR, I will do it)

**Comment by DavideD** [comment]
Rebased and Merged https://github.com/hibernate/hibernate-ogm/pull/1069

Thanks a lot, I did a few changes to the test though

**Pull request closed without merging**